pkg/proxy/register.go

```diff
@@ -116,6 +119,26 @@ func (p *Register) createDirector(proxyDefinition *Definition, balancer Balancer
 			}
 		}
 
+		paramNames := paramNameExtractor.Extract(path)
+		for _, paramName := range paramNames {
```

Comment:
This should probably be encapsulated into a function. Maybe something like `applyParameters` or anything on the same line of thinking. I don't know where this function should go, maybe the `paramNameExtractor`?

Reply:
Then I will put it in a function in this same file, I would like to leave the `paramNameExtractor` unaware of request, context and so on.
We already have functions playing with the path in here like: 
```
cleanSlashes @ pkg/proxy/register.go:178 
singleJoiningSlash @ pkg/proxy/register.go:193
```

Sounds good?

Reply:
Yep that's ok